Hi,

linux/drivers/watchdog/orion5x_wdt.c cannot be compiled because of
undefined symbol, ORION5X_TCLK.
Here is a patch to compile it.

Thanks.
-- 
Kunihiko IMAI

--- linux-2.6.28-rc9.orig/drivers/watchdog/orion5x_wdt.c	2008-12-21
08:34:10.000000000 +0900
+++ linux-2.6.28-rc9/drivers/watchdog/orion5x_wdt.c	2008-12-21
15:08:23.000000000 +0900
@@ -29,12 +29,13 @@
 #define  WDT_EN			0x0010
 #define WDT_VAL			(TIMER_VIRT_BASE + 0x0024)

-#define WDT_MAX_DURATION	(0xffffffff / ORION5X_TCLK)
+extern int orion5x_tclk;
+#define WDT_MAX_DURATION	(0xffffffff / orion5x_tclk)
 #define WDT_IN_USE		0
 #define WDT_OK_TO_CLOSE		1

 static int nowayout = WATCHDOG_NOWAYOUT;
-static int heartbeat =  WDT_MAX_DURATION;	/* (seconds) */
+static int heartbeat =  -1;	/* (seconds) */
 static unsigned long wdt_status;
 static spinlock_t wdt_lock;

@@ -45,7 +46,7 @@
 	spin_lock(&wdt_lock);

 	/* Set watchdog duration */
-	writel(ORION5X_TCLK * heartbeat, WDT_VAL);
+	writel(orion5x_tclk * heartbeat, WDT_VAL);

 	/* Clear watchdog timer interrupt */
 	reg = readl(BRIDGE_CAUSE);
@@ -87,7 +88,7 @@
 static int orion5x_wdt_get_timeleft(int *time_left)
 {
 	spin_lock(&wdt_lock);
-	*time_left = readl(WDT_VAL) / ORION5X_TCLK;
+	*time_left = readl(WDT_VAL) / orion5x_tclk;
 	spin_unlock(&wdt_lock);
 	return 0;
 }
@@ -216,6 +217,9 @@
 	spin_lock_init(&wdt_lock);

 	ret = misc_register(&orion5x_wdt_miscdev);
+
+	if ( heartbeat < 0 )
+		heartbeat =  WDT_MAX_DURATION;
 	if (ret == 0)
 		printk("Orion5x Watchdog Timer: heartbeat %d sec\n",
 								heartbeat);

-------------------------------------------------------------------
List admin: http://lists.arm.linux.org.uk/mailman/listinfo/linux-arm-kernel
FAQ:        http://www.arm.linux.org.uk/mailinglists/faq.php
Etiquette:  http://www.arm.linux.org.uk/mailinglists/etiquette.php
